src/lib/schema.js

```diff
@@ -39,6 +39,12 @@ const bucketKeys = {
     outgoingBytes: (b, t) => `s3:buckets:${t}:${b}:outgoingBytes`,
 };
 
+const metric = {
+    stateKey: (k, b) => k.slice('s3:buckets:'.length).replace(`${b}:`, ''),
+    timestampKey: (k, b) =>
+        k.slice('s3:buckets:0000000000000:'.length).replace(`${b}:`, ''),
```

Comment:
I fear this is a bit fragile to rely on the number of digits in a timestamp. Since fields look to be delimited by colons, and since you must know how many fields you want to truncate, can you instead slice up to the number of fields you want to remove, where they are separated by colons? You can consider using a regexp if that makes the thing simpler.